### ISSUE TITLE
Test failures: Int overflow on 32 bits arches

### ISSUE THREAD

**Issue by eclipseo** [prometheus/procfs#319]
Go 1.15 rc 1 on Fedora Rawhide armv7h and i686:

```
# github.com/prometheus/procfs [github.com/prometheus/procfs.test]
./proc_maps_test.go:67:4: constant 93905347534848 overflows uintptr
./proc_maps_test.go:68:4: constant 93905347543040 overflows uintptr
./proc_maps_test.go:76:4: constant 93905347579904 overflows uintptr
./proc_maps_test.go:77:4: constant 93905347584000 overflows uintptr
./proc_maps_test.go:85:4: constant 93905365065728 overflows uintptr
./proc_maps_test.go:86:4: constant 93905365200896 overflows uintptr
./proc_maps_test.go:94:4: constant 140598276243456 overflows uintptr
./proc_maps_test.go:95:4: constant 140598291931136 overflows uintptr
./proc_maps_test.go:103:4: constant 140598291931136 overflows uintptr
./proc_maps_test.go:104:4: constant 140598292082688 overflows uintptr
./proc_maps_test.go:104:4: too many errors
FAIL	github.com/prometheus/procfs [build failed]
```

**Comment by discordianfish** [comment]
This happens with current master? I though https://github.com/prometheus/procfs/pull/314 should have fixed this.

**Comment by eclipseo** [comment]
I am packaging the latest released version 0.1.3 / 2020-06-14. I will try with a patch from master.

**Comment by eclipseo** [comment]
Ok it works with x86_32 now. And armv7hl fails with another reason.
